src/stirling/utils/linux_headers_test.cc

```diff
@@ -28,116 +29,13 @@ namespace px {
 namespace stirling {
 namespace utils {
 
-namespace {
-std::string GetPathToTestDataFile(const std::string_view fname) {
-  constexpr char kTestDataBasePath[] = "src/stirling/utils/testdata";
-  return testing::BazelRunfilePath(std::filesystem::path(kTestDataBasePath) / fname);
-}
-}  // namespace
-
 using ::px::testing::TempDir;
 using ::testing::EndsWith;
 
-bool operator==(const KernelVersion& a, const KernelVersion& b) {
+bool operator==(const system::KernelVersion& a, const system::KernelVersion& b) {
   return (a.version == b.version) && (a.major_rev == b.major_rev) && (a.minor_rev == b.minor_rev);
 }
 
-// A list of kernel version identification methods that use /proc only.
-// Used in a number of tests below.
-const std::vector<KernelVersionSource> kProcFSKernelVersionSources = {
-    KernelVersionSource::kProcVersionSignature, KernelVersionSource::kProcSysKernelVersion};
-
-TEST(LinuxHeadersUtils, LinuxVersionCode) {
-  KernelVersion version{4, 18, 1};
-  EXPECT_EQ(version.code(), 266753);
-}
-
-TEST(LinuxHeadersUtils, ParseKernelVersionString) {
-  EXPECT_OK_AND_EQ(ParseKernelVersionString("4.18.0-25-generic"), (KernelVersion{4, 18, 0}));
-  EXPECT_OK_AND_EQ(ParseKernelVersionString("4.18.4"), (KernelVersion{4, 18, 4}));
-  EXPECT_NOT_OK(ParseKernelVersionString("4.18."));
-  EXPECT_NOT_OK(ParseKernelVersionString("linux-4.18.0-25-generic"));
-}
-
-TEST(LinuxHeadersUtils, GetKernelVersionFromUname) {
-  StatusOr<KernelVersion> kernel_version_status = FindKernelVersion({KernelVersionSource::kUname});
-  ASSERT_OK(kernel_version_status);
-  KernelVersion kernel_version = kernel_version_status.ValueOrDie();
-
-  // We don't know on what host this test will run, so we don't know what the version code will be.
-  // But we can put some bounds, to check for obvious screw-ups.
-  // We assume test will run on a Linux machine with kernel 3.x.x or higher,
-  // and version 9.x.x or lower.
-  // Yes, we're being very generous here, but we want this test to pass the test of time.
-  EXPECT_GE(kernel_version.code(), 0x030000);
-  EXPECT_LE(kernel_version.code(), 0x090000);
-}
-
-TEST(LinuxHeadersUtils, GetKernelVersionFromNoteSection) {
-  StatusOr<KernelVersion> kernel_version_status =
-      FindKernelVersion({KernelVersionSource::kVDSONoteSection});
-  ASSERT_OK(kernel_version_status);
-  KernelVersion kernel_version = kernel_version_status.ValueOrDie();
-
-  // We don't know on what host this test will run, so we don't know what the version code will be.
-  // But we can put some bounds, to check for obvious screw-ups.
-  // We assume test will run on a Linux machine with kernel 3.x.x or higher,
-  // and version 9.x.x or lower.
-  // Yes, we're being very generous here, but we want this test to pass the test of time.
-  EXPECT_GE(kernel_version.code(), 0x030000);
-  EXPECT_LE(kernel_version.code(), 0x090000);
-}
```

Comment:
[src/stirling/utils/BUILD.bazel](https://github.com/pixie-io/pixie/blob/1eb7873ed8263f0223bef1b87ff20c0b35432d7b/src/stirling/utils/BUILD.bazel#L78-L80) is still enabled to use full qemu emulation, but it seems like we moved the VDSO specific test to `src/common/system`. Did you investigate if we could remove that tag from here?

Reply:
I believe full qemu emulation is unneeded here. I've removed the tag in the most recent commit.